lib/components-schema-v1_0_x.ts

```diff
@@ -40,7 +336,18 @@ export const componentsDefinitionSchema_v1_0_x = {
                     icon: { type: 'string', description: 'Icon shown for component in Digital Editor' },
                     properties: {
                         type: 'array',
-                        items: { type: 'string' },
+                        items: {
+                            oneOf: [
+                                {
+                                    type: 'string',
+                                },
+                                {
+                                    type: 'object',
+                                    additionalProperties: false,
```

Comment:
seems you forgot about required: ['name', 'label', 'control', 'dataType'] ?

Reply:
except none are required. Whether it becomes a valid object depends. If name is set, it will be merged with the matching property in componentProperties.

This might end up requiring some additional validation.